LISA/Executor.cpp

```diff
@@ -233,24 +233,25 @@ void Executor::doInstall(std::string type,
         setProgress(progress, OperationStage::DOWNLOADING);
     };
 
-    Downloader downloader{url, progressListener};
+    auto tmpFilePath = tmpDirPath + extractFilename(url);
+
+    {
+        Downloader downloader{url, progressListener};
 
-    auto downloadSize = downloader.getContentLength();
-    auto tmpFreeSpace = Filesystem::getFreeSpace(tmpDirPath);
+        auto downloadSize = downloader.getContentLength();
+        auto tmpFreeSpace = Filesystem::getFreeSpace(tmpDirPath);
 
-    INFO("download size: ", downloadSize, " free tmp space: ", tmpFreeSpace);
+        INFO("download size: ", downloadSize, " free tmp space: ", tmpFreeSpace);
 
-    if (downloadSize > tmpFreeSpace) {
-        std::string message = std::string{} + "not enough space on " + tmpPath + " (available: "
+        if (downloadSize > tmpFreeSpace) {
+            std::string message = std::string{} + "not enough space on " + tmpPath + " (available: "
                 + std::to_string(tmpFreeSpace) +", required: " + std::to_string(downloadSize) + ")";
-        throw std::runtime_error(message);
+            throw std::runtime_error(message);
+        }
+        Filesystem::File tmpFile{tmpFilePath};
```

Comment:
tmpFile is only needed for download, so I suggest to modify Downloader.get to accept tmpFilePath and create tmpFile inside the function

Reply:
Done